File: ts-sdk/src/sdk/models/operations/gettermsofservices.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";


export enum GetTermsOfServicesTosTypeEnum {
    External = "external",
    Managed = "managed"
}


export class GetTermsOfServicesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=tos_type" })
  tosType?: GetTermsOfServicesTosTypeEnum;
}


export class GetTermsOfServicesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  queryParams: GetTermsOfServicesQueryParams;
}


export class GetTermsOfServicesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  termsOfServices?: shared.TermsOfServices;
}

File: go-sdk/pkg/models/operations/gettermsofservices.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetTermsOfServicesTosTypeEnum string

const (
	GetTermsOfServicesTosTypeEnumExternal GetTermsOfServicesTosTypeEnum = "external"
	GetTermsOfServicesTosTypeEnumManaged  GetTermsOfServicesTosTypeEnum = "managed"
)

type GetTermsOfServicesQueryParams struct {
	TosType *GetTermsOfServicesTosTypeEnum `queryParam:"style=form,explode=true,name=tos_type"`
}

type GetTermsOfServicesRequest struct {
	Retries     *utils.RetryConfig
	QueryParams GetTermsOfServicesQueryParams
}

type GetTermsOfServicesResponse struct {
	ClientError     *shared.ClientError
	ContentType     string
	StatusCode      int64
	TermsOfServices *shared.TermsOfServices
}
